src/Platform/Microsoft.Testing.Platform/OutputDevice/TargetFrameworkParser.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Globalization;
+
+namespace Microsoft.Testing.Platform.OutputDevice;
+
+internal static class TargetFrameworkParser
+{
+    internal /* for testing */ static string? GetShortTargetFramework(string frameworkDescription)
+    {
+        // https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation.frameworkdescription?view=net-8.0
+        string netFramework = ".NET Framework";
+        if (frameworkDescription.StartsWith(netFramework, ignoreCase: false, CultureInfo.InvariantCulture))
+        {
+            // .NET Framework 4.7.2
+            if (frameworkDescription.Length < (netFramework.Length + 6))
+            {
+                return frameworkDescription;
+            }
+
+            char major = frameworkDescription[netFramework.Length + 1];
+            char minor = frameworkDescription[netFramework.Length + 3];
+            char patch = frameworkDescription[netFramework.Length + 5];
+
+            if (major == '4' && minor == '6' && patch == '2')
+            {
+                return "net462";
+            }
+            else if (major == '4' && minor == '7' && patch == '1')
+            {
+                return "net471";
+            }
+            else if (major == '4' && minor == '7' && patch == '2')
+            {
+                return "net472";
+            }
+            else if (major == '4' && minor == '8' && patch == '1')
+            {
+                return "net481";
+            }
+            else
+            {
+                // Just return the first 2 numbers.
+                return $"net{major}{minor}";
```

Comment:
Let's hope we won't have some more releases :)

Reply:
If there are more it will have to evolve, there is no logic in when I would expect and not expect seeing short version. ;/